runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -53,6 +57,18 @@
 @Slf4j
 public class AddressSpaceView extends AbstractView {
 
+    public static long removalCauseCacheSize = 500_000;
+
+    /**
+     * This cache maintains debug information that is required to
+     * determine the cause of a NoRollbackException. Using ehcache's
+     * deepSizeOf it shows that the memory foot print of the
+     * removalCauseCache is around ~50mb.
+     */
+    public static Cache<Long, RemovalCause>  removalCauseCache = CacheBuilder.newBuilder()
```

Comment:
`public static final Cache<Long, RemovalCause>  REMOVAL_CAUSE_CACHE ...`
I really don't like it. Can we use maybe metrics for that somehow? @StrikingThirteen what do you think?

Reply:
Great, propose a better approach.